### PR TITLE
[Merged by Bors] - chore(txs): mempool optimization

### PR DESCRIPTION
**Pull request by acud**
## Motivation

Improves some aspects of the mempool handling.

## Test Plan

Tests passing

## TODO

- [ ] Explain motivation or link existing issue(s)
- [ ] Test changes and document test plan
- [ ] Update documentation as needed
- [ ] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

**Comment by acud** [comment]
bors try

File: txs/cache.go [hunk 1]
```diff
@@ -553,15 +553,10 @@ func (c *Cache) cleanupAccounts(accounts ...types.Address) {
...
-//     conservative cache is conservative in that it only counts principal's spending for pending transactions.
-//     a tx rejected due to insufficient balance MAY become feasible after a layer is applied (principal
-//     received incoming funds). when we receive a errInsufficientBalance tx, we should store it in db and
-//     re-evaluate it after each layer is applied.
 //   - errTooManyNonce: when a principal has way too many nonces, we don't want to blow up the memory. they should
 //     be stored in db and retrieved after each earlier nonce is applied.
 func acceptable(err error) bool {
-	return err == nil || errors.Is(err, errInsufficientBalance) || errors.Is(err, errTooManyNonce)
+	return err == nil || errors.Is(err, errTooManyNonce)
 }
```
Comment (fasmat) [inline txs/cache.go:559]: Could you explain in the PR description from a users perspective what this change will accomplish?

File: txs/cache.go [hunk 2]
```diff
@@ -115,7 +116,6 @@ func (ac *accountCache) precheck(logger *zap.Logger, ntx *NanoTX) (*list.Element
 		break
 	}
 	if balance < ntx.MaxSpending() {
-		ac.moreInDB = true
```
Comment (fasmat) [inline txs/cache.go:118]: What effect does the removal of this have?

**Comment by acud** [comment]
bors try

**Review by fasmat — APPROVED** [review]

**Comment by acud** [comment]
bors merge

**Comment by acud** [comment]
bors cancel

**Review by ivan4th — APPROVED** [review]

**Comment by acud** [comment]
bors merge

**Comment by acud** [comment]
bors merge

**Pull request closed without merging**